chromium_src/components/favicon/core/large_icon_service_impl.cc

```diff
@@ -0,0 +1,12 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "components/favicon/core/large_icon_service_impl.h"
+
+#define server_url_(URL) \
+  server_url_("https://favicons.proxy.brave.com/faviconV2")
```

Comment:
this will break upstream components unit tests. I'd go something like this:
```cpp
namespace favicon {
class ServerUrlSource {
  static std::string_view Get() {
    if (base::CurrentTestVendor::Get() == base::TestVendor::kChromium) {
      return kGoogleServerV2Url;
    } else {
      return "https://favicons.proxy.brave.com/faviconV2";
    }
  }
}
```

```diff
class BASE_EXPORT CurrentTestVendor {
 private:
  FRIEND_TEST_ALL_PREFIXES(CurrentTestVendorTest, IsBrave);
+ friend favicon::ServerUrlSource;
  friend variations::PublicKeyWrapper;
  static TestVendor Get();
};
```

```cpp
#define server_url_(URL) server_url_(ServerUrlSource::Get())
```